### ISSUE TITLE
Use oauth user's twiiter/github username as default as opposed to random strings

### ISSUE THREAD

**Issue by Mayowa-Ojo** [benawad/dogehouse#1457]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
This is just a suggestion. Currently when a user is authorized via oauth, a random username is generated. I think it'd be more intuitive to use the user's **twitter** username or **github** username as default. 

**Describe alternatives you've considered**

**Additional context**

**Comment by ghost** [comment]
This might be problematic if someone signed up with one platform and someone with the same username on a different platform tries to sign up.
Would it be better to prompt the user to choose an username?

**Comment by Mayowa-Ojo** [comment]
Yes you're right. I also thought of this and the first plausible solution I imagine is to append a string of short random generated numbers to the username if it's already taken. E.g if `unorthodev` is taken, `unorthodev-1355` could be generated for the user. A real-world example of this is [Quora](https://quora.com)

**Comment by ajhalili2006** [comment]
I would like also to suggest to prompt the user to sign in to existing account so these OAuth accounts can be associated with single DogeHouse account if the user wants to.

Otherwise, the user may opt to create a new account instead with some random math suffix or contact DogeHouse Support.
